### PR TITLE
report compute units without tx_wide_compute_cap feature enabled

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem

Need to report bpf program compute units to cost model without having to enable `tx_wide_compute_cap` feature. 

#### Summary of Changes

Fixes #

### CODE REVIEW

File: program-runtime/src/invoke_context.rs [hunk 1]
```diff
@@ -280,10 +280,6 @@ impl<'a> InvokeContext for ThisInvokeContext<'a> {
             }
             self.current_compute_budget = compute_budget;
 
-            if !self.feature_set.is_active(&tx_wide_compute_cap::id()) {
-                self.compute_meter = ComputeMeter::new_ref(self.current_compute_budget.max_units);
```
Comment (jackcmay) [inline program-runtime/src/invoke_context.rs:284]: This resets the compute meter for each instruction, without this we would be tracking the meter tx-wide.  I think in order to track tx wide but cap per instruction we would need to track it independently

Comment (taozhu-chicago) [reply program-runtime/src/invoke_context.rs:284]: Looks like compute units are accumulated to `timings` per instruction (https://github.com/solana-labs/solana/blob/master/runtime/src/message_processor.rs#L121-L125). Though this reset causes `post_remaining_units` to be resolved into `max_units` again, maybe to get `post_remaining_units` differently?

File: runtime/src/message_processor.rs [hunk 2]
```diff
@@ -117,7 +117,8 @@ impl MessageProcessor {
             invoke_context.pop();
 
             time.stop();
-            let post_remaining_units = compute_meter.borrow().get_remaining();
+            let post_remaining_units = invoke_context.get_compute_meter().borrow().get_remaining();
```
Comment (taozhu-chicago) [inline runtime/src/message_processor.rs:120]: think this is a bug - when `meter` being reset during `invokeContext.push()`, a new refcell<compute_meter> is created, and consumed. Should use this new one for `post`. Doing some testing now

**Comment by Lichtso** [comment]
@taozhu-chicago Can you rebase it onto master? I think my recent refactoring causes merge conflicts.

File: runtime/src/message_processor.rs [hunk 3]
```diff
@@ -104,9 +102,13 @@ impl MessageProcessor {
             }
 
             invoke_context.set_instruction_index(instruction_index);
+            let mut pre_remaining_units = 0u64;
             let result = invoke_context
                 .push(message, instruction, program_indices, None)
                 .and_then(|_| {
+                    // read the `pre` value after potential meter reset, right before processing
+                    pre_remaining_units =
+                        invoke_context.get_compute_meter().borrow().get_remaining();
```
Comment (jackcmay) [inline runtime/src/message_processor.rs:111]: We could read the post-units in this `and_then` and do the calc here as well, might make it a bit tidier.
```
units consumed = 0
push and then
read pre units
process instruction
read post units
units-consumed = post - pre
...
accumulate (program_id, time, units_consumed)
```

File: runtime/src/message_processor.rs (hunk 2 above)
Comment (jackcmay) [reply runtime/src/message_processor.rs:122]: Yeah, but not a bug, was not originally meant to be used without the tx-wide feature

**Comment by taozhu-chicago** [comment]
> @taozhu-chicago Can you rebase it onto master? I think my recent refactoring causes merge conflicts.

nice refactor ... rebased!

File: runtime/src/message_processor.rs (hunk 3 above)
Comment (taozhu-chicago) [reply runtime/src/message_processor.rs:107]: good idea! moved timer and unit measure into and_then, closer to process instruction.

**Review by jackcmay — APPROVED** [review]

**Pull request merged**